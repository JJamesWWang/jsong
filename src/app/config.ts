const baseProtocol = "https";
const baseUrl = "jsong-api.herokuapp.com";

export const websocketEndpoint = (username: string) =>
  // @ts-ignore
  baseProtocol === "http"
    ? `ws://${baseUrl}/ws/${username}`
    : `wss://${baseUrl}/ws/${username}`;

export const claimHostEndpoint = (uid: string) =>
  `${baseProtocol}://${baseUrl}/lobby/host/${uid}`;

export const setPlaylistEndpoint = `${baseProtocol}://${baseUrl}/lobby/playlist`;

export const startGameEndpoint = (uid: string) =>
  `${baseProtocol}://${baseUrl}/lobby/start/${uid}`;

export const setReadyEndpoint = (uid: string) =>
  `${baseProtocol}://${baseUrl}/lobby/ready/${uid}`;

export const trackEndpoint = (round: number) => `${baseProtocol}://${baseUrl}/lobby/track?round=${round}`;

export const endGameEndpoint = (uid: string) =>
  `${baseProtocol}://${baseUrl}/lobby/end/${uid}`;
