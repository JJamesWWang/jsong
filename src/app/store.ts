import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import counterReducer from "../features/counter/counterSlice";
import lobbyReducer from "../features/lobby/lobbySlice";
import chatReducer from "../features/chat/chatSlice";

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    lobby: lobbyReducer,
    chat: chatReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
