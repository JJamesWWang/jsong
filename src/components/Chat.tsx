import useWebSocket from "react-use-websocket";
import { websocketEndpoint } from "../app/config";
import { useAppSelector } from "../app/hooks";
import MessageBox from "./MessageBox";
import InputBox from "./ui/InputBox";

function Chat() {
  const member = useAppSelector((state) => state.lobby.member);

  const { sendJsonMessage } = useWebSocket(
    websocketEndpoint(member ? member.username : ""),
    { share: true }
  );

  return (
    <>
      <MessageBox />
      <InputBox name="chat" autoFocus={true} onSubmit={sendJsonMessage} />
    </>
  );
}

export default Chat;
