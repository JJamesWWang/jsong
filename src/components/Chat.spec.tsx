import { screen, render, waitFor } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../app/store";
import Chat from "./Chat";

window.HTMLElement.prototype.scrollIntoView = function () {};

async function renderComponent() {
  return await waitFor(() =>
    render(
      <Provider store={store}>
        <Chat />
      </Provider>
    )
  );
}

describe("chat", () => {
  it("renders a message box and input box", async () => {
    const { container } = await renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("has a chat label", async () => {
    await renderComponent();
    const chatText = screen.getByText("Chat:");
    expect(chatText).toBeInTheDocument();
  });
});
