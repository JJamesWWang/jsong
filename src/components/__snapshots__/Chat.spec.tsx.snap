// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat renders a message box and input box 1`] = `
<div>
  <p>
    Chat:
  </p>
  <ul
    class="ul"
  >
    <li />
  </ul>
  <form
    autocomplete="off"
    class="form"
    name="chat"
  >
    <input
      class="input"
      name="chat"
      size="32"
      type="text"
    />
  </form>
</div>
`;
